phoenix-core/src/main/java/org/apache/phoenix/hbase/index/Indexer.java

```diff
@@ -800,5 +804,62 @@ public static void enableIndexing(HTableDescriptor desc, Class<? extends IndexBu
     properties.put(Indexer.INDEX_BUILDER_CONF_KEY, builder.getName());
     desc.addCoprocessor(Indexer.class.getName(), null, priority, properties);
   }
+
+    /**
+     * stop indexer when region split, but split failed, and then rollback, open region.
+     * this need init builder and writer,recoveryWriter.
+     *
+     * @param env
+     */
+  private void initBuilderAndWriter(RegionCoprocessorEnvironment env) {
```

Comment:
this is a part of start(), extract init builder and writer, because region split, builder and writer are close. but split failed, and then reigon rollback,  reopen region, so need init builder and writer again.